### ISSUE TITLE
[WIP] Implementation of visual testing epic

### ISSUE THREAD

**Issue by corwintines** [ethereum/ethereum-org-website#13043]
_This is part of our [Testing](https://github.com/ethereum/ethereum-org-website/issues/12746) epic from our Q2 roadmap._

## Description

We have implemented support for storybook to do testing on [ethereum.org](https://ethereum.org/). We would like to be able to catch visual regressions before they reach production, helping us build confidence so that we can improve stability of ethereum.org through a testing suite. The first phase of this will be implementing visual storybook tests on our suite of components. 

Below is the list of components in the ethereum.org repo, broken down by [Atomic design](https://www.uxpin.com/studio/blog/atomic-ui-components/#:~:text=Atoms.,Examples%20include%20buttons%20or%20forms.).

## How to contribute
If you’d like to help with adding stories for components, please follow these steps:

1. Leave a comment here asking which component you would liek to work out. We will be going from the top of the list to bottom (Atoms to Pages).
2. We assign you to the file (by adding your handle next to the file name in the list).
    - Read and follow our storybook implementation guide for some tips & guidance.
    - **If you were assigned to multiple files** then create a separate PR for each component. It will make the testing and review process much simpler and more organized.
3. Once you finish it (PR merged), we’ll mark it as done.
4. Repeat!

-----

### Atoms

In Atomic Design, as in chemistry, atoms are the basic elements that help inform everything. In the world of web applications, atoms are the foundational elements, such as HTML tags, fonts, animations, and color palettes. Web design “atoms” can also be less concrete. Examples include buttons or forms.

```[tasklist]
### Typeography
- [x] Headings
- [x] @chakra-ui/components/Link
- [x] Text 
```

```[tasklist]
### Media & icons - All visual references used in ethereum.org
- [x] Avatars
- [x] Icons
- [ ] Logo
- [ ] MergeInfoGraphic
- [ ] TranslationChartImage
- [ ] Trilema
```

```[tasklist]
### Forms - Base html components used in forms
- [x] Buttons
- [x] Checkbox
- [x] Input variations and state
- [x] Radio
- [x] Select
- [x] Slider
- [x] Switch
```

### Molecules

Molecules are the next-largest building block. Created by the joining of different atomic elements, molecules are complex by nature. Because they’re the product of various atoms, though, it’s possible to break them down, conceptually, into UI elements that are easier to digest. Examples of web design molecules include the things that become the backbone of the larger design system, such as form labels or input field.

```[tasklist]
### Navigation - components used to navigate to other pages. Inner or external to ethereum.org.
- [ ] BannerNotification
- [ ] DismissableBanner (BugBountyBanner, make sub issue for this)
- [x] Breadcrumbs
- [ ] CallToContribute
- [ ] Contributors
- [ ] DeveloperDocsLinks
- [ ] DocLink
- [ ] DocsNav
- [ ] LeftNavBar
- [x] Links
- [ ] MergeArticleList
- [ ] TableOfContents
```

```[tasklist]
### Display content - components used to show data/information primary
- [ ] ActionCard
- [ ] AssetDownload (Make sure this tests AssetDownloadArtist and AssetDownloadImage, expand in issue)
- [ ] BannerGrid (Expand in issue)
- [ ] BeaconChainActions
- [ ] BoxGrid
- [ ] BugBountyCards
- [ ] Callout
- [ ] Card
- [ ] CardList
- [ ] Codeblock
- [ ] CommunityEvents
- [ ] ContributorsQuizBanner
- [ ] CopyToClipboard
- [ ] DataProductCard
- [ ] EnergyConsumptionChart
- [ ] EnvWarningBanner
- [ ] EventCard
- [ ] FeedbackCard
- [ ] FileContributors
- [ ] GitStars
- [ ] HorizontalCard
- [ ] InfoBanner
- [ ] IssuesList
- [ ] Layer2ProductCard
- [ ] Leaderboard
- [ ] LearningToolsCardGrid
- [x] GlossaryDefinition
- [ ] NetworkUpgradeSummary
- [ ] Layer2Onboard
- [ ] MarkdownImage
- [ ] MatomoOptOut
- [ ] MdComponents (expand in separate issue)
- [ ] OrderedList
- [ ] ProductCard
- [ ] ProductDisclaimer
- [ ] ProductList
- [ ] RandomAppList
- [ ] Progress
- [ ] Quiz (expand in separate issue)
- [ ] RoadmapActionCard
- [ ] RoadmapImageContent
- [ ] SocialListItem
- [ ] StablecoinBoxGrid
- [ ] StablecoinsTable
- [x] Stat
- [ ] StateErrorMessage
- [x] Tables
- [ ] Tags (expand in issue)
- [ ] TitleCardList
- [ ] TranslationBanner
- [ ] TranslationBannerLegal
- [ ] TranslationLeaderboard
- [ ] TutorialMetadata
- [ ] TutorialTags
- [ ] UpgradeStatus
- [ ] UpgradeTableOfContents
- [ ] Youtube
```

```[tasklist]
### Disclosure content - components used to hide and show content useing interactive functions
- [x] Accordion
- [ ] ButtonDropdown
- [ ] StablecoinAccordion
- [x] Switch
- [ ] Tabs
```

```[tasklist]
### Overlay content - components used on top from the main page
- [ ] CodeModal
- [ ] ExpandableCard
- [ ] ExpandableInfo
- [x] FeedbackWidget
- [x] GlossaryTooltip
- [x] Modal
- [ ] Popover
- [x] Tooltip
- [ ] TooltipLink
```

```[tasklist]
### Action feedback - components used to show feedback of a user action
- [x] Alerts
```

### Organisms

Atoms combine to form molecules, and groups of molecules form organisms. In the world of Atomic Design, organisms are the UI elements that shape both the appearance and functionality of a website. They’re also the elements that start to impact user interface. The way a developer arranges molecules informs the site experience and the complexity of the finished product. Examples of organisms include logos, search fields, and main navigation which together may form a header organism.

```[tasklist]
- [ ] Header (break this down in another issue)
- [ ] Hero (break this out into another issue)
- [ ] Footer
- [ ] MainArticle
```

```[tasklist]
### Snippets - composed components has use cases or references to display the same type of data
-  [ ] Simulator (break this out into another issue)
```

### Templates

At this phase of the Atomic Design process, we start to break with the chemistry analogy and shift back into the lexicon of front-end development, as a whole. Templates, then, are “organisms” strung together at the page-level or beyond. Templates, online atoms, organisms, and molecules, are highly concrete. They provide a fixed context for the more abstract pieces to fit and are responsible for pulling the site together into something resembling its final form. An HTML wireframe is an excellent example of a template.

```[tasklist]
### Templates - base pages to be used in creation of new content. They incorporate all of the components mandatory for the page
- [ ] Docs
- [ ] Roadmap [Blocked on refactor]
- [ ] RootLayout (BaseLayout)
- [ ] Staking [Blocked on refactor]
- [ ] Static
- [ ] Tutorial
- [ ] Upgrade [Blocked on refactor]
- [ ] Usecases [Blocked on refactor]
```

### Pages

Pages, finally, are the final element of Atomic Design. According to Frost himself, pages are the specific instances of templates. Pages are the most tangible element of all and are the places users spend most of their time. They’re also one of the most essential phases of the Atomic Design process since the final iteration of pages is where developers get to see whether the entire design system is effective or not. In short, the final appearance of the pages dictates whether the product design is ready to launch, or whether the developer needs to loop back and make changes to earlier UI design elements.

```[tasklist]
### Special cases - Pages that are unique and require custom made components
- [ ] FindWallet (break out into issue)
- [ ] Staking (break out into issue)
```

```[tasklist]
### Pages - List of pages with final content to showcase and inspire
- [ ] / (homepage)
- [ ] /404
- [ ] /assets
- [ ] /bug-bounty
- [ ] /community
- [ ] /contributing/translation-program
- [ ] /contributing/translation-program/acknowledgements
- [ ] /contributing/translation-program/contributors
- [ ] /dapps
- [ ] /developers
- [ ] /developers/learning-tools
- [ ] /developers/local-environment
- [ ] /developers/tutorials
- [ ] /eth
- [ ] /gas
- [ ] /get-eth
- [ ] /layer-2
- [ ] /learn
- [ ] /quizzes
- [ ] /roadmap
- [ ] /roadmap/vision
- [ ] /run-a-node
- [ ] /stablecoins
- [ ] /staking
- [ ] /staking/deposit-contract
- [ ] /wallets
- [ ] /wallets/find-wallet
- [ ] /what-is-ethereum
```

-----

## How to contribute
If you’d like to help with adding stories for components, please follow these steps:

1. Leave a comment here asking which component you would liek to work out. We will be going from the top of the list to bottom (Atoms to Pages).
2. We assign you to the file (by adding your handle next to the file name in the list).
    - Read and follow our storybook implementation guide for some tips & guidance.
    - **If you were assigned to multiple files** then create a separate PR for each component. It will make the testing and review process much simpler and more organized.
3. Once you finish it (PR merged), we’ll mark it as done.
4. Repeat!

**Comment by TommWolf** [comment]
For me it does not matter on which component I will work. I simply want to help.

**Comment by corwintines** [comment]
Awesome, thanks @TommWolf!

I created an issue for the Logo component: https://github.com/ethereum/ethereum-org-website/issues/13067

If you comment in this issue, ill assign you :)

**Comment by TylerAPfledderer** [comment]
@corwintines noting here that I have #12960 up for `RootLayout` (now becoming `BaseLayout`), and looking at building a consolidated layout as discussed previously with Pablo and Nuno.

**Comment by wackerow** [comment]
Started attaching the sub-task Issues as "milestones" for this epic ([13043](https://github.com/ethereum/ethereum-org-website/milestone/6)) to help track

**Comment by pettinarip** [comment]
Hey @corwintines could you assign me the `TableOfContents`? wip #13156

**Comment by corwintines** [comment]
@pettinarip created an issue and assigned. Will update description over there.

**Comment by Baystef** [comment]
Hi @corwintines , I'll like to be assigned the MergeInfoGraphic story. Thanks

**Comment by tomasgrusz** [comment]
Hey, would love to help out with this, happy to start working on Form components (e.g. Buttons, Inputs, Checkbox,..) one by one.

**Comment by tomasgrusz** [comment]
> > Hey, would love to help out with this, happy to start working on Form components (e.g. Buttons, Inputs, Checkbox,..) one by one.
> 
> Hey @tomasgrusz ! Those components already have stories. They are shown as checked and only here for the organization of the list :)

My mistake @TylerAPfledderer. In that case, I'm happy to pick up any of the remaining Atom components so that we can have all of them done.

**Comment by corwintines** [comment]
@tomasgrusz I pinged you in #13067 just now. If you are open for it, will assign :)

**Comment by Baystef** [comment]
@corwintines I'll like to be assigned the TranslationChartImage story

**Comment by Baystef** [comment]
@wackerow @corwintines  I'll like to work on the CallToContribute component under Molecules.

**Comment by corwintines** [comment]
Creating those issues now for you @Baystef. Thank you for your patience and taking those on!

**Comment by Baystef** [comment]
@corwintines Please assign Contributors and DeveloperDocsLinks to me. Thanks

**Comment by prxjwal** [comment]
@corwintines I would like to work on BannerNotification pls assign

**Comment by corwintines** [comment]
Will create the issue for you now @prxjwal

**Comment by fernandoguadmendezespinoza17** [comment]
Ok

**Comment by fernandoguadmendezespinoza17** [comment]
Ayuda 🆘⛑️

**Comment by fernandoguadmendezespinoza17** [comment]
Hello

**Comment by GovardhaneNitin** [comment]
Hi! I'm interested in working on this issue. I'd like to implement the Storybook component for BannerNotification.

**Comment by alexandriaroberts** [comment]
Hi @corwintines , I am just seeing this.😊 Are we still going through these? If we are, while you create some other issue on adding products can I please do DocsNav, LeftNavBar and MergeArticleList? Thank you!

**Comment by JoeChenJ** [comment]
Hey @corwintines , I would like to help with this issue. Can I start with `BugBountyCards`?

Thanks!

**Comment by Afeter8** [comment]
![layer-2-hub-hero.jpg](https://github.com/user-attachments/assets/116324e8-8fca-4c62-ae38-541c9059e6c5)

![participate.png](https://github.com/user-attachments/assets/90ad8c61-62a1-499c-ad41-23fb486e6781)

![garden.jpg](https://github.com/user-attachments/assets/f2846e8b-ec71-4239-9b6f-df1e5dd98bed)

![guides-hub-hero.jpg](https://github.com/user-attachments/assets/cde6d1a0-2e1c-4cdd-84b1-59d45f73a329)

![roadmap-hub-hero.jpg](https://github.com/user-attachments/assets/4b4c3780-a1cb-4ceb-a27f-abc7d6e56716)

![learn-hub-hero.png](https://github.com/user-attachments/assets/c2c4afd2-fc74-4311-815d-9ff573c3a66d)

![quizzes-hub-hero.png](https://github.com/user-attachments/assets/18d9ea7f-f28f-47b4-86b2-3b89b05d60bd)

![developers-hub-hero.jpg](https://github.com/user-attachments/assets/5b357eb5-3359-4928-9ea4-47fb4732ca7a)

![local-communities.png](https://github.com/user-attachments/assets/f2d13ac5-b07c-41c8-abe6-06d6e53f21e9)

![community-hero.png](https://github.com/user-attachments/assets/8c8f7524-a4e9-4a38-bb47-2037a2ca5c98)

![hackathon_transparent.png](https://github.com/user-attachments/assets/f3bf68ea-a755-4783-89ca-f7d8db083242)

![future_transparent.png](https://github.com/user-attachments/assets/99bbd770-b910-422e-8bd2-a3c61fcbf281)

![finance_transparent.png](https://github.com/user-attachments/assets/f0f8d9ea-4ccb-4bde-86eb-6e0ab1ea3ebc)

![favicon.png](https://github.com/user-attachments/assets/9e6488bb-b2d1-4af6-bad7-4ad44f5f8c44)

![hero-panda.png](https://github.com/user-attachments/assets/32822524-c794-4a13-9558-434bbd63906c)

![hero(1).png](https://github.com/user-attachments/assets/9c11bb97-7caf-4498-a87e-4aaa11e89e55)

![merge-panda.png](https://github.com/user-attachments/assets/dcd81057-21f0-407f-9816-8968b645687c)

![impact_transparent.png](https://github.com/user-attachments/assets/fa04cf20-3535-4087-8568-e79e147f47e8)

![panda-ops-banner.png](https://github.com/user-attachments/assets/4a34b801-ac29-420d-86ed-5b875957549b)

**Comment by Afeter8** [comment]
![people-learning.png](https://github.com/user-attachments/assets/baaf390f-f8c3-4148-a919-04bbf5d0a5d3)

![robot-help-bar.png](https://github.com/user-attachments/assets/291725bd-eb79-4f06-9124-905011220a7a)

![roadmap-future.png](https://github.com/user-attachments/assets/ce38ab74-ca57-4cc1-b9e8-599c95e2c879)

![roadmap-transactions.png](https://github.com/user-attachments/assets/6a190e63-6fe0-494c-ad8b-65f2e0eff1e3)

![roadmap-ux.png](https://github.com/user-attachments/assets/96108e1d-2236-4195-89cb-5c2eea57a826)

![solidity-banner.png](https://github![roadmap-security.png](https://github.com/user-attachments/assets/ee6031f3-27bc-42d9-a5ec-edf6d0b86acd)

ff9f-4add-9de8-e4a6df80a544)

![Cargando roadmap-security.png …]()

![roadmap-main.png](https://github.com/user-attachments/assets/d93452ec-26f5-48e6-81db-4d5859563aa3)

![man-baby-woman.png](https://github.com/user-attachments/assets/541aa069-4fe3-43f8-86eb-72c55efb9fc5)

![panda-ops-banner.png](https://github.com/user-attachments/assets/ddeafec8-d0c0-4d7c-aacd-ee457d808e47)

![man-and-dog-playing.png](https://github.com/user-attachments/assets/6dcedb97-7f2a-4767-b376-86c0be9aa4d5)

![mainnet.png](https://github.com/user-attachments/assets/6dbee101-7d65-4897-8273-c9b8574ea89a)

![magicians.png](https://github.com/user-attachments/assets/d1f931d1-2e7d-4faf-a8ea-d318da989ce1)

![rollup-2.png](https://github.com/user-attachments/assets/e9f2c820-6f74-489f-a09b-0b00fe4bd52c)

![infrastructure_transparent.png](https://github.com/user-attachments/assets/00e19d17-2c0f-490a-9b52-a1d08ff9fd47)

**Comment by Afeter8** [comment]
![poap(2).png](https://github.com/user-attachments/assets/14dc464b-7f67-44fc-8775-95038e9bfe0d)

**Comment by Afeter8** [comment]
[https://www.cloudns.net.]
(https://www.cloudns.net.)

**Comment by Afeter8** [comment]
Hello

**Comment by Afeter8** [comment]
Hello

**Comment by abdallh12345** [comment]
abdallh

**Comment by Mamadkin** [comment]
عالی متمعن خیلی خوب است
